src/nvim/api/buffer.c

```diff
@@ -86,6 +86,12 @@ void buffer_del_line(Buffer buffer, Integer index, Error *err)
   FUNC_ATTR_DEFERRED
 {
   Array array = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if(index >= buf->b_ml.ml_line_count) {
+    return;
```

Comment:
Hmm..., a race condition could cause us to try and delete line 99 after 98 was deleted by something else. Not a problem for this PR, but maybe we should have a way for clients to lock buffers.